Applications/Spire/Source/Ui/EditableTableView.cpp

```diff
@@ -262,6 +274,32 @@ EditableTableView::EditableTableView(
   get_header().get_item(0)->set_is_resizeable(false);
   get_header().get_widths()->set(0, scale_width(26));
   set_style(*this, TABLE_VIEW_STYLE());
+  setFocusProxy(&get_body());
+  get_body().installEventFilter(this);
+  m_current_connection = get_current()->connect_update_signal(
+    std::bind_front(&EditableTableView::on_current, this));
+}
+
+bool EditableTableView::eventFilter(QObject* watched, QEvent* event) {
+  if(watched == &get_body()) {
+    if(event->type() == QEvent::KeyPress) {
+      auto& key_event = *static_cast<QKeyEvent*>(event);
+      if(key_event.key() == Qt::Key_Shift) {
```

Comment:
What is special about the shift key?

Reply:
When pressing Shift + Tab to do Backtab navigation, the Shift key triggers the RANG selection mode in the TableBody. The RANGE selection mode causes a crash, so this code prevents the Shift key from triggering the RANG selection.

Reply:
Yes. Another reason is that I don't think Shift + Tab should change the selection mode.

Reply:
Please do not introduce hacks and workarounds to mask issues.